pkg/cmd/run/watch/watch.go

```diff
@@ -0,0 +1,228 @@
+	opts := &WatchOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+		Now:        time.Now,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "watch <run-selector>",
+		Short: "Runs until a run completes, showing its progress",
+		Annotations: map[string]string{
+			"IsActions": "true",
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// support `-R, --repo` override
+			opts.BaseRepo = f.BaseRepo
+
+			if len(args) > 0 {
+				opts.RunID = args[0]
+			} else if !opts.IO.CanPrompt() {
+				return &cmdutil.FlagError{Err: errors.New("run ID required when not running interactively")}
+			} else {
+				opts.Prompt = true
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+
+			return watchRun(opts)
+		},
+	}
+	cmd.Flags().BoolVar(&opts.ExitStatus, "exit-status", false, "Exit with non-zero status if run fails")
+	cmd.Flags().IntVarP(&opts.Interval, "interval", "i", defaultInterval, "Refresh interval in seconds")
+
+	return cmd
+}
+
+func watchRun(opts *WatchOptions) error {
+	c, err := opts.HttpClient()
+	if err != nil {
+		return fmt.Errorf("failed to create http client: %w", err)
+	}
+	client := api.NewClientFromHTTP(c)
+
+	repo, err := opts.BaseRepo()
+	if err != nil {
+		return fmt.Errorf("failed to determine base repo: %w", err)
+	}
+
+	runID := opts.RunID
+	var run *shared.Run
+
+	if opts.Prompt {
+		cs := opts.IO.ColorScheme()
+		runs, err := shared.GetRunsWithFilter(client, repo, 10, func(run shared.Run) bool {
+			return run.Status != shared.Completed
+		})
+		if err != nil {
+			return fmt.Errorf("failed to get runs: %w", err)
+		}
+		if len(runs) == 0 {
+			return fmt.Errorf("found no in progress runs to watch")
+		}
+		runID, err = shared.PromptForRun(cs, runs)
+		if err != nil {
+			return err
+		}
+		// TODO silly stopgap until dust settles and PromptForRun can just return a run
+		for _, r := range runs {
+			if fmt.Sprintf("%d", r.ID) == runID {
+				run = &r
+				break
+			}
+		}
+	} else {
+		run, err = shared.GetRun(client, repo, runID)
+		if err != nil {
+			return fmt.Errorf("failed to get run: %w", err)
+		}
+	}
+
+	if run.Status == shared.Completed {
+		return nil
+	}
+
+	prNumber := ""
+	number, err := shared.PullRequestForRun(client, repo, *run)
+	if err == nil {
+		prNumber = fmt.Sprintf(" #%d", number)
+	}
+
+	if runtime.GOOS == "windows" {
+		opts.IO.EnableVirtualTerminalProcessing()
+	}
+	// clear entire screen
+	fmt.Fprintf(opts.IO.Out, "\x1b[2J")
+
+	annotationCache := map[int][]shared.Annotation{}
+
+	duration, err := time.ParseDuration(fmt.Sprintf("%ds", opts.Interval))
+	if err != nil {
+		return fmt.Errorf("could not parse interval: %w", err)
+	}
+
+	for run.Status != shared.Completed {
+		run, err = renderRun(*opts, client, repo, run, prNumber, annotationCache)
+		if err != nil {
+			return err
+		}
+		time.Sleep(duration)
+	}
+
+	if opts.ExitStatus && run.Conclusion != shared.Success {
+		return cmdutil.SilentError
+	}
+
+	return nil
```

Comment:
I wonder if it would be nice to output a message to the user stating the run completed and its status (success or failure)?